src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/ExceptionAggregator.cs

```diff
@@ -165,7 +165,7 @@ private static bool ThrowAnOCE(Exception ex, CancellationState cancellationState
             // See QueryTaskGroupState.WaitAll for the main plinq exception handling logic.
 
             // check for co-operative cancellation.
-            OperationCanceledException cancelEx = ex as OperationCanceledException;
+            OperationCanceledException? cancelEx = ex as OperationCanceledException;
```

Comment:
Nit: this could have also used an is, just combining the two below blocks together, e.g.
```C#
if (ex is OperationCanceledException cancelEx)
{
    if (cancelEx.CancellationToken == cancellationState.ExternalCancellationToken &&
        cancellationState.ExternalCancellationToken.IsCancellationRequested)
    {
        return true;  // let the OCE(extCT) be rethrown.
    }

    if (cancelEx.CancellationToken == cancellationState.MergedCancellationToken &&
        cancellationState.MergedCancellationToken.IsCancellationRequested &&
        cancellationState.ExternalCancellationToken.IsCancellationRequested)
    {
        return true; // convert internal cancellation back to OCE(extCT).
    }
}
```